ICSharpCode.Decompiler/CSharp/TranslatedExpression.cs

```diff
@@ -227,6 +227,9 @@ public TranslatedExpression ConvertTo(IType targetType, ExpressionBuilder expres
 					.WithRR(convAnnotation.ConversionResolveResult)
 					.ConvertTo(targetType, expressionBuilder, checkForOverflow, allowImplicitConversion);
 			}
+			if (Expression is ThrowExpression && allowImplicitConversion) {
+				return this; // Throw expressions have no type and are implicitly convertible to any type
```

Comment:
Returning `this` may be problematic: the `ResolveResult` annotation on `this` still has the old type, which may cause assertions due to stack type mismatches.

Reply:
This is actually OK, there's no requirement that `ConvertTo(targetType, allowImplicitConversion=true).Type.Equals(targetType)`.
(such a requirement only exists for `allowImplicitConversion=false`, and even then the types only have to be equal after type erasure)